doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1154 @@
+    "\n",
+    "# Function to transform a pandas dataframe\n",
+    "def transform_df(the_df: pd.DataFrame) -> pd.DataFrame:\n",
+    "    df = the_df.copy()\n",
+    "    \n",
+    "    df[\"trip_duration\"] = (df[\"dropoff_at\"] - df[\"pickup_at\"]).dt.seconds\n",
+    "    df = df[df[\"trip_duration\"] > 60]\n",
+    "    df = df[df[\"trip_duration\"] < 24 * 60 * 60] \n",
+    "    df.drop([\"dropoff_at\", \"pickup_at\", \"pickup_location_id\", \"fare_amount\"]\n",
+    "            , axis=1, inplace=True)\n",
+    "    df[\"dropoff_location_id\"] = df[\"dropoff_location_id\"].fillna(-1)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28814e79-f53e-4527-a959-e952f7d57fe5",
+   "metadata": {},
+   "source": [
+    "## Define a Trainable (callable) function <a class=\"anchor\" id=\"define_trainable\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "0bd4b4a0-bcc8-49c3-8b5e-2590f9100a9f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "sklearn: 1.1.2\n"
+     ]
+    }
+   ],
+   "source": [
+    "# import standard sklearn libraries\n",
+    "import sklearn\n",
+    "from sklearn.base import BaseEstimator\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.linear_model import LinearRegression\n",
+    "from sklearn.tree import DecisionTreeRegressor\n",
+    "from sklearn.metrics import mean_absolute_error\n",
+    "print(f\"sklearn: {sklearn.__version__}\")\n",
+    "\n",
+    "# import ray AIR libraries\n",
+    "from ray import air, tune\n",
+    "from ray.air import session\n",
+    "from ray.train.sklearn import SklearnCheckpoint\n",
+    "from ray.train.batch_predictor import BatchPredictor\n",
+    "from ray.train.sklearn import SklearnPredictor\n",
+    "\n",
+    "# set global random seed for sklearn models\n",
+    "np.random.seed(415)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cdb3c9a4-b2df-4063-8c04-f5e0136c90bf",
+   "metadata": {},
+   "source": [
+    "<b>Define a \"Trainable\" object that you can pass into a Tune run.</b> Ray Tune has two ways of defining a trainable, namely the [Function API](https://docs.ray.io/en/latest/tune/api_docs/trainable.html#trainable-docs) and the Class API. Both are valid ways of defining a trainable, but *the Function API is generally recommended*.\n",
+    "\n",
+    "Let’s say we want to optimize a simple objective function like a (x ** 2) + b in which a and b are the hyperparameters we want to tune to minimize the objective. Since the objective also has a variable x, we need to test for different values of x. Given concrete choices for a, b and x we can evaluate the objective function and get a score to minimize.\n",
+    "\n",
+    "**In the cell below, our \"Trainable\" function is called `train_model`**.  It takes as input a config dictionary argument, and returns a simple dict output. "
```

Comment:
we should also describe what is the purpose of reporting metrics (talking back to Tune, providing incremental results if called in a loop)